circus-train-avro/src/main/java/com/hotels/bdp/circustrain/avro/util/FileSystemPathResolver.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hotels.bdp.circustrain.avro.util;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+import static org.springframework.util.StringUtils.isEmpty;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DFSConfigKeys;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.bdp.circustrain.api.CircusTrainException;
+
+public class FileSystemPathResolver {
+  private static final Logger LOG = LoggerFactory.getLogger(FileSystemPathResolver.class);
+
+  private final Configuration configuration;
+
+  public FileSystemPathResolver(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  public Path resolveScheme(Path path) {
+    try {
+      URI uri = path.toUri();
+      if (isEmpty(uri.getScheme())) {
+        String scheme = FileSystem.get(configuration).getScheme();
+        Path result = new Path(new URI(scheme, uri.getUserInfo(), uri.getHost(), uri.getPort(), uri.getPath(),
+            uri.getQuery(), uri.getFragment()).toString());
+        LOG.info("Added scheme {} to path {}. Resulting path is {}", scheme, path, result);
+        return result;
+      }
+    } catch (URISyntaxException | IOException e) {
```

Comment:
Do we need either/both of these? If neither then we can probably get rid of the whole try/catch? Or just remove one of the exception types?

Reply:
URISyntaxException is needed for new URI(....) with the new scheme and IOException for getting the FileSystem

Reply:
Ah yes, of course, the new URI() is where it comes from, makes sense.